File: src/main/java/com/jackmeng/stl/stl_ListenerDeployable.java
// Copyright 2023 Jack Meng. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package com.jackmeng.stl;

public interface stl_ListenerDeployable<A>
    extends stl_Callback<Void, Iterable<A>>
{
  default Void call(Iterable<A> e)
  {
    return (Void) null;
  }

  default void empty(int length)
  {

  }
}

File: src/main/java/com/jackmeng/stl/stl_ListenerDeployable.kt
// Copyright 2023 Jack Meng. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.
package com.jackmeng.stl

import com.jackmeng.stl.stl_Callback
import java.lang.Void

interface stl_ListenerDeployable<A>:stl_Callback<Void? , Iterable<A>?>
{
	override fun call(e:Iterable<A>):Void
	{
		return null as Void
	}
	
	fun empty(length:Int)
	{
	}
}